### ISSUE TITLE
Squiz.Commenting.FunctionComment doesn't work with variadic arguments

### ISSUE THREAD

**Issue by tfrommen** [squizlabs/PHP_CodeSniffer#1652]
**Example Code**:

```php
<?php
/**
 * @param string[] ...$names Names.
 *
 * @return void
 */
function foo( string ...$names ) {

}
```

When I run this:

```sh
phpcs test.php --standard=Squiz --sniffs=Squiz.Commenting.FunctionComment
```

I get the following output:

```text
---------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
---------------------------------------------------------------------------------
 7 | ERROR | Expected type hint "array"; found "string" for ...$names
---------------------------------------------------------------------------------
```

However, `string` is the correct type hint for the **variadic** argument `$names`, which is documented as array of strings (i.e., `string[]`).

Also, I'm not sure I understand why a **comment**-specific sniff is arguing about the **type hint** and not the _incorrect_ annotation.

**Comment by webimpress** [comment]
It's is not clear what way is correct. Please see:
https://youtrack.jetbrains.com/issue/WI-29429
https://github.com/phpDocumentor/fig-standards/issues/121

PHP_CodeSniffer suggest to use type of single element with variadic param documentation.
I can see that PHPStorm generates always "array type" instead. Personally I would prefer single element type in docs. "Array type" i s only inside the function.

**Comment by gsherwood** [comment]
Reading those linked issues, the current thinking is that this code should be documented like this:
```
<?php
/**
 * @param string ...$names Names.
 *
 * @return void
 */
function foo( string ...$names ) {

}
```
Which is what the sniff wants you to do. Until such a time as this is cleared up, I'm happy to leave the sniff reporting the same errors as it is doing now.
